### ISSUE TITLE
$query store prototype

### ISSUE THREAD

**Issue by zyxd** [sveltetools/svelte-pathfinder#4]
Because a $query store have String prototype there may be property name conflicts.

```javascript
import { query } from 'svelte-pathfinder'

console.log($query.search) // function String.search by default
$query.search = 'test'
console.log($query.search) // string 'test'
```

Maybe need to rewrite a $query store to an object with a toString() function that returns '?one=1&second=2&...' string?

**Comment by PaulMaly** [comment]
Yea, I know about this thing. The same behavior with `$path` store and its `pattern()` method. The main reason was to simplify these things as possible. Perhaps, we should find some more elegant solution.

**Comment by PaulMaly** [comment]
Partially fixed in 3.0.0. Now `query` and `path` params bring together in single `.params` property and seem it should exclude name conflicts described in this issue. Also, `pattern()` is not an injected method of `path` value, but separate derived reactive function. Please, read the CHANGELOG and README for more info.
